nilearn/datasets/struct.py

```diff
@@ -14,23 +14,23 @@
 from ..image import get_data, new_img_like, resampling
 from ._utils import fetch_files, get_dataset_descr, get_dataset_dir
 
-_package_directory = os.path.dirname(os.path.abspath(__file__))
-MNI152_FILE_PATH = os.path.join(
-    _package_directory,
-    "data",
-    "mni_icbm152_t1_tal_nlin_sym_09a_converted.nii.gz",
+_package_directory = Path(__file__).absolute().parent
+MNI152_FILE_PATH = (
+    _package_directory
+    / "data"
+    / "mni_icbm152_t1_tal_nlin_sym_09a_converted.nii.gz"
 )
-GM_MNI152_FILE_PATH = os.path.join(
-    _package_directory,
-    "data",
-    "mni_icbm152_gm_tal_nlin_sym_09a_converted.nii.gz",
+GM_MNI152_FILE_PATH = (
+    _package_directory
+    / "data"
+    / "mni_icbm152_gm_tal_nlin_sym_09a_converted.nii.gz"
 )
-WM_MNI152_FILE_PATH = os.path.join(
-    _package_directory,
-    "data",
-    "mni_icbm152_wm_tal_nlin_sym_09a_converted.nii.gz",
+WM_MNI152_FILE_PATH = (
+    _package_directory
+    / "data"
+    / "mni_icbm152_wm_tal_nlin_sym_09a_converted.nii.gz"
 )
-FSAVERAGE5_PATH = os.path.join(_package_directory, "data", "fsaverage5")
+FSAVERAGE5_PATH = _package_directory / "data" / "fsaverage5"
```

Comment:
I quickly checked and those constants are "kinda public": they are in the __init__.py of the datasets subpackage but they are NOT in our documentation, so even though this may be an issue for some users who could rely on them being string I think this change is still OK with no deprecation cycle.